### PR TITLE
adding support for Azure Billing

### PR DESCRIPTION
**Pull request by selvamt94**

### CODE REVIEW

File: charts/core/README.md [hunk 1]
```diff
@@ -235,6 +235,13 @@ Parameter | Description | Default | Notes
 `awsbilling.image.repository` | csp adapter image repository | `neuvector/neuvector-csp-adapter` | Follow AWS subscription instruction
 `awsbilling.image.tag` | csp adapter image tag | `1.0.0` | Follow AWS subscription instruction
 `awsbilling.image.imagePullPolicy` | csp adapter image pull policy | `IfNotPresent` | Follow AWS subscription instruction
+`global.azurebilling.enabled` | If true, install Azure billing csp adapter | `false` | **Note**: default admin user is disabled when global.azurebilling enabled, use configmap to create admin-role user to manage NeuVector deployment.
+`global.azurebilling.annotations` | Annotaion for csp adapter  | `nil` | Follow Azure subscription instruction
+`global.azurebilling.imagePullSecrets` | Pull secret for csp adapter image | `nil` | Follow Azure subscription instruction
+`global.azurebilling.image.repository` | csp adapter image repository | `neuvector/neuvector-csp-adapter` | Follow Azure subscription instruction
```
Comment (KeithMnemonic) [inline charts/core/README.md:241]: If we do not want to maintain a seperate fork or branch for building the CNAB for Azure, we need to follow these guidelines.  https://learn.microsoft.com/en-us/partner-center/marketplace/azure-container-technical-assets-kubernetes?tabs=linux#update-the-helm-chart

File: charts/core/templates/csp-deployment.yaml [hunk 2]
```diff
@@ -26,6 +26,10 @@ spec:
       imagePullSecrets:
         - name: {{ .Values.awsbilling.imagePullSecrets }}
       {{- end }}
+      {{- if .Values.global.azurebilling.imagePullSecrets }}
+      imagePullSecrets:
+        - name: {{ .Values.global.azurebilling.imagePullSecrets }}
+      {{- end }}
       containers:
```
Comment (KeithMnemonic) [inline charts/core/templates/csp-deployment.yaml:33]: If we do not want to maintain a seperate fork or branch for building the CNAB for Azure, we need to follow these guidelines. https://learn.microsoft.com/en-us/partner-center/marketplace/azure-container-technical-assets-kubernetes?tabs=linux#update-the-helm-chart

we need to add in some extra env variables here or using somethibg like

        # envFrom:
        #   - configMapRef:
        #       name: neuvector-csp-pod
        env:

File: charts/core/values.yaml [hunk 3]
```diff
@@ -15,7 +15,23 @@ leastPrivilege: false
 global: # required for rancher authentication (https://<Rancher_URL>/)
   cattle:
     url:
-
+  azurebilling:
```
Comment (KeithMnemonic) [inline charts/core/values.yaml:18]: We need to follow the naming structure Microsoft dictates for the CNAB

https://learn.microsoft.com/en-us/partner-center/marketplace/azure-container-technical-assets-kubernetes?tabs=linux#update-the-helm-chart

**Review by KeithMnemonic — CHANGES_REQUESTED** [review]
Suggested updates if we want to simplify maintenance of the Charts needed to build the CNAB used to create the Azure offer:

See:  https://learn.microsoft.com/en-us/partner-center/marketplace/azure-container-technical-assets-kubernetes?tabs=linux#update-the-helm-chart

**Comment by garyduan** [comment]
I suggest we move the AWS billing parameters to 'global' as well.

File: charts/core/values.yaml (hunk 3 above)
Comment (KeithMnemonic) [reply charts/core/values.yaml:18]: we need to change `azurebilling` to `azure`

**Comment by KeithMnemonic** [comment]
@selvamt94 I tested this with the neuvector2 plan and it deployed successfully and I can see it billed.

2023-09-20T18:42:35.239|INFO|CSPBillingAdapter|Attempt a billing cycle update
2023-09-20T18:42:35.241|INFO|CSPBillingAdapter|Filtering records for 2023-09-20T17:42:32.076781+00:00 <= reporting_time < 2023-09-20T18:42:32.076781+00:00
2023-09-20T18:42:35.241|INFO|CSPBillingAdapter|Metric: 'managed_node_count', average = 3, records: [{'base_product': 'cpe:/o:suse:neuvector:5.2.0', 'managed_node_count': 3, 'reporting_time': '2023-09-20T18:22:36.472852+00:00'}]
2023-09-20T18:42:35.241|INFO|CSPBillingAdapter|Billable usage: {'managed_node_count': 5}, records: [{'base_product': 'cpe:/o:suse:neuvector:5.2.0', 'managed_node_count': 3, 'reporting_time': '2023-09-20T18:22:36.472852+00:00'}]
2023-09-20T18:42:35.241|INFO|CSPBillingAdapter|Metric: managed_node_count=5, Volume dimension: 5
2023-09-20T18:42:35.241|INFO|CSPBillingAdapter|Determined billable dimensions {'tier_1': 5} for usage {'managed_node_count': 5}
2023-09-20T18:42:36.386|INFO|CSPBillingAdapter|New metered billing record added with ID 37b01ec7-67b9-4129-9401-d343385ae2c2:
2023-09-20T18:42:36.386|INFO|CSPBillingAdapter|Metering submitted, record_id: {'tier_1': {'record_id': '37b01ec7-67b9-4129-9401-d343385ae2c2', 'status': 'submitted'}}

**Comment by jeremy-moffitt** [comment]
@selvamt70 Keith and Yoga tell me that they've tested these changes and are happy with them. What else is needed to get this PR merged?

**Review by brett060102 — APPROVED** [review]
Change LGTM and has passed testing by Keith and Yoga's testing.

**Review by KeithMnemonic — APPROVED** [review]
LGTM

**Pull request merged**